src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/AvTrace.cs

```diff
@@ -258,73 +259,66 @@ internal static bool IsDebuggerAttached()
         //  note: labels start at index 1, parameters start at index 0
         //
 
-        public string Trace( TraceEventType type, int eventId, string message, string[] labels, object[] parameters )
+        public string Trace(TraceEventType type, int eventId, string message, string[] labels, params Span<object> parameters)
         {
             // Don't bother building the string if this trace is going to be ignored.
 
-            if( _traceSource == null
-                || !_traceSource.Switch.ShouldTrace( type ))
-            {
+            if(_traceSource == null|| !_traceSource.Switch.ShouldTrace(type))
                 return null;
-            }
-
 
             // Compose the trace string.
 
             AvTraceBuilder traceBuilder = new AvTraceBuilder(AntiFormat(message)); // Holds the format string
-            ArrayList arrayList = new ArrayList(); // Holds the combined labels & parameters arrays.
+            List<object> combinedList = new(parameters.Length * 2); // Holds the combined labels & parameters arrays.
 
             int formatIndex = 0;
 
-            if (parameters != null && labels != null && labels.Length > 0)
+            if (!parameters.IsEmpty && labels?.Length > 0)
             {
                 int i = 1, j = 0;
-                for( ; i < labels.Length && j < parameters.Length; i++, j++ )
+                for (; i < labels.Length && j < parameters.Length; i++, j++)
                 {
                     // Append to the format string a "; {0} = '{1}'", where the index increments (e.g. the second iteration will
                     // produce {2} & {3}).
 
-                    traceBuilder.Append("; {" + (formatIndex++).ToString() + "}='{" + (formatIndex++).ToString() + "}'" );
+                    traceBuilder.Append($"; {{{++formatIndex}}}='{{{++formatIndex}}}'");
```

Comment:
Changing `formatIndex++` to `++formatIndex` will change the output string from `; {0}='{1}'` to `; {1}='{2}`. This seems like an unintentionally-introduced bug.

Reply:
Yeah, thanks, don't know how I've managed to get that there. Fixed.

Reply:
Good catch